src/display/text_layer.js

```diff
@@ -46,23 +47,43 @@ const DEFAULT_FONT_SIZE = 30;
 const DEFAULT_FONT_ASCENT = 0.8;
 const ascentCache = new Map();
 
-function getAscent(fontFamily, ctx) {
+function getCtx(size, isOffscreenCanvasSupported, alpha = true) {
+  let ctx;
+  if (isOffscreenCanvasSupported && FeatureTest.isOffscreenCanvasSupported) {
+    ctx = new OffscreenCanvas(size, size).getContext("2d", { alpha });
```

Comment:
Do we "need" to use `isOffscreenCanvasSupported` here, i.e. are there any clear benefits?
If this is indeed necessary, maybe do that in a *separate commit* (in this PR) to reduce the size of these changes such that reviewing becomes slightly easier?

Reply:
Also, do we actually need to use different `alpha` values or is that perhaps just how the code happened to be written originally here?
If not, I'd suggest that we simply hard-code `alpha: false` since that's what we use in e.g. the viewer in general.

Reply:
> Do we "need" to use `isOffscreenCanvasSupported` here, i.e. are there any clear benefits? If this is indeed necessary, maybe do that in a _separate commit_ (in this PR) to reduce the size of these changes such that reviewing becomes slightly easier?

I found this benchmark:
https://www.measurethat.net/Benchmarks/Show/12247/2/setfont-measuretext-offscreencanvas-vs-canvas

which shows that the `OffscreenCanvas` is slightly faster.
I ran one myself in adding an extra loop (with 100 iterations) around:
https://github.com/mozilla/pdf.js/pull/15722/files#diff-eb5220436f55f0c6ff687c15781981b323be818cb77e49d53d66f7f372f1f23dR502-R506

In using an `OffscreenCanvas`, the loop around `layout` call is 10% faster (mean = 75ms, stddev=11 when it's 82ms/9 for a canvas).

Reply:
My guess is that setting a font should be faster because we don't have to get some default style properties from the document itself.

Reply:
Thanks for the context; I'd agree that a 10% improvement enough to warrant some additional code.